### PR TITLE
First draft of image basics notebook and associated demo image

### PR DESCRIPTION
**Pull request by SolarDrew**
Notebook lesson going over:
- Concept of images as arrays
- How to display image arrays with `imshow()`
- Colour maps and colour bars
- Adjusting range of plotted values
- Changing axis limits and origin
- RGB image arrays
- `plt.subplots()`
- Slicing images
- Animating images

There should also be room for a colour map rant if Stu doesn't mind me stealing it.

### CODE REVIEW

File: images-and-plotting-instructor.ipynb [hunk 1]
```diff
@@ -0,0 +1,617 @@
...
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# Get some import statements out of the way.\n",
+    "from __future__ import division, print_function\n",
+    "%matplotlib inline\n",
+    "from matplotlib import pyplot as plt\n",
```
Comment (Cadair) [inline images-and-plotting-instructor.ipynb]: we need to standardise on either:

``` python
from matplotlib import pyplot as plt
```

or

``` python
import matplotlib.pyplot as plt
```

personally I prefer the latter.

File: images-and-plotting-instructor.ipynb [hunk 2]
```diff
@@ -0,0 +1,617 @@
...
+    "# Display the image with a better colour map.\n",
+    "plt.imshow(moon, cmap='gray')\n",
+    "plt.colorbar()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The full list of available colour maps (in matplotlib 1.5 - more have been added in 2.0, and the defaults have changed) can be found [here](http://matplotlib.org/examples/color/colormaps_reference.html)."
```
Comment (Cadair) [inline images-and-plotting-instructor.ipynb]: All the new ones were added in 1.5, the defaults will not change until 2.0

File: images-and-plotting-instructor.ipynb [hunk 3]
```diff
@@ -0,0 +1,617 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "plt.imshow(moon, cmap='gray', origin='lower', extent=[-1, 1, 0, 2])"
```
Comment (Cadair) [inline images-and-plotting-instructor.ipynb]: Also maybe cover `interpolation='nearest'` here?

File: images-and-plotting-instructor.ipynb [hunk 4]
```diff
@@ -0,0 +1,617 @@
...
+   "source": [
+    "plt.imshow(moon, cmap='gray', vmin=75, vmax=150)\n",
+    "plt.colorbar()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As you can see, this allows us to increase the contrast of the image at the cost of discounting extreme values, or we can include a broader range of values but see less detail."
```
Comment (Cadair) [inline images-and-plotting-instructor.ipynb]: is it worth mentioning `norm=` i.e. log scaling etc?

File: images-and-plotting-instructor.ipynb (hunk 1 above)
Comment (SolarDrew) [inline images-and-plotting-instructor.ipynb]: Fair enough. Any particular reason? So I can decide why I disagree :P

File: images-and-plotting-instructor.ipynb [hunk 5]
```diff
@@ -0,0 +1,617 @@
...
+    "%matplotlib nbagg\n",
+    "\n",
+    "fig, ax = plt.subplots()\n",
+    "display = plt.imshow(gal)\n",
+    "\n",
+    "titles = ['Red component', 'Green component', 'Blue component', 'Combined image']\n",
+    "cmaps = ['Reds_r', 'Greens_r', 'Blues_r']\n",
+    "\n",
+    "def animate(i):\n",
+    "    try:\n",
```
Comment (Cadair) [inline images-and-plotting-instructor.ipynb]: instead of the try except block you could index with `i%gal.shape[2]`??

**Comment by Cadair** [comment]
looks good!

**Comment by SolarDrew** [comment]
Cool. I'll try and get most of those changes done tonight and we'll merge it in the morning.

File: images-and-plotting-instructor.ipynb (hunk 5 above)
Comment (SolarDrew) [inline images-and-plotting-instructor.ipynb]: Yes, but then it doesn't show the combined image, which I would prefer to.

**Comment by SolarDrew** [comment]
Got a few more minor cosmetic changes on the way, please don't merge yet, anyone.

File: images-and-plotting-instructor.ipynb [hunk 6]
```diff
@@ -0,0 +1,718 @@
...
+   "outputs": [],
+   "source": [
+    "# Output the image minimum, mean and maximum.\n",
+    "print('Image min:', moon.min(), '; Image mean:', moon.mean(), '; Image max: ', moon.max())\n",
+    "\n",
+    "# Output the array dtype.\n",
+    "print('Data type:', moon.dtype)\n",
+    "\n",
+    "# Output image resolution.\n",
+    "print('Image resolution:', moon.shape)"
```
Comment (dpshelio) [inline images-and-plotting-instructor.ipynb]: I would argue this is not resolution, but size of the image.

File: images-and-plotting-instructor.ipynb [hunk 7]
```diff
@@ -0,0 +1,718 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This tells us that the image has a resolution of 512 x 512 pixels, and is stored as integers between 0 and 255. This is a common way of normalising images, but they can just as easily be stored as floats between 0 and 1. More commonly with astronomical data though, an image will consist of photon counts (i.e. integers), so the minimum will be 0 and there may be no upper limit to possible values. "
```
Comment (dpshelio) [inline images-and-plotting-instructor.ipynb]: I believe every CCD has an upper value, one due to saturation, and the other due to the digitalisation of the image, 8bit, 16,... etc.

File: images-and-plotting-instructor.ipynb [hunk 8]
```diff
@@ -0,0 +1,718 @@
...
+    "\n",
+    "There is no single 'best' colour map - different colour maps display different kinds of image most clearly - but the `jet` map is almost never an appropriate choice for displaying any data. In general, colour maps which vary luminosity uniformly (such as the `'gray'` colour map above or the `'cubehelix'` colour map) tend to be better. Plots of various colour maps' luminosities can be found [here](http://matplotlib.org/users/colormaps.html).<br/><br/>\n",
+    "\n",
+    "For a good background on this topic and a description of a decent all-round colour map scheme, see [this paper](http://www.kennethmoreland.com/color-maps/ColorMapsExpanded.pdf).\n",
+    "\n",
+    "</section>"
+   ]
+  },
```
Comment (dpshelio) [inline images-and-plotting-instructor.ipynb]: This block is not rendered by github... (?) [Or I don't see it](https://github.com/drewleonard42/workshop_sunpy_astropy/blob/plotting-images/images-and-plotting-instructor.ipynb)

File: images-and-plotting-instructor.ipynb [hunk 9]
```diff
@@ -0,0 +1,718 @@
...
+    "<h3><span class=\"fa fa-pencil\"></span> Load and plot an image </h3>\n",
+    "</div>\n",
+    "\n",
+    "<ol>\n",
+    "    <li> Try loading and plotting some other image arrays from `skimage.data`. Choose one of these images and print some basic information about the values it contains.</li>\n",
+    "    <li> Plot your chosen image with `imshow()`. Apply a colour map of your choice and display a colour bar.</li>\n",
+    "</ol>\n",
+    "</section>"
+   ]
+  },
```
Comment (dpshelio) [inline images-and-plotting-instructor.ipynb]: and neither this one

File: images-and-plotting-instructor.ipynb [hunk 10]
```diff
@@ -0,0 +1,718 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# Load the Hubble image from fig/galaxy.jpg\n",
+    "gal = plt.imread('fig/galaxy.jpg')\n",
```
Comment (dpshelio) [inline images-and-plotting-instructor.ipynb]: I will call the variable `galaxy` or even  `galaxy_image`

File: images-and-plotting-instructor.ipynb [hunk 11]
```diff
@@ -0,0 +1,718 @@
...
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# 1\n",
+    "\n",
+    "myim = data.coffee()\n",
```
Comment (dpshelio) [inline images-and-plotting-instructor.ipynb]: `myim` is not very meaningful name...

**Comment by dpshelio** [comment]
@drewleonard42 looks great! sorry I commented now, just one day before!!

**Pull request merged**